### PR TITLE
Enable zk with out kafka cluster

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Bugfix

### Description

Use reflection hackery to force `enableEntityConfigControllerCheck` to false. 

### Additional Context

We can't configure SCRAM users before starting Kafka without this.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Review by showuon — APPROVED** [review]
LGTM! I agree this is the best workaround, compared with using `KafkaZkClient.apply` with some other hack with the private `zkTlsClientAuthEnabled` method, or invoking ConfigCommand to add scram users. I think this did what we want to test, and ZK is going to get removed, we don't have to spend too much time on it. Thanks for the fix!

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ public class KafkaClusterConfig {
     @Singular
     private final Map<String, String> brokerConfigs;
 
-    @SuppressWarnings("java:S5738") // silence warnings about the use of deprecated code
+    @SuppressWarnings({ "java:S5738", "removal" }) // silence warnings about the use of deprecated code
```
Comment (k-wall) [inline impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java:144]: I didn't know about "removal"

File: impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java [hunk 2]
```diff
@@ -491,4 +493,22 @@ private void createScramUsersInZookeeper(List<UserScramCredentialRecord> parsedC
...
+            final Class<? extends KafkaZkClient> zkClientClass = zkClient.getClass();
+            final Field enableEntityConfigControllerCheck = zkClientClass.getDeclaredField("enableEntityConfigControllerCheck");
+            enableEntityConfigControllerCheck.setAccessible(true);
+            enableEntityConfigControllerCheck.setBoolean(zkClient, false);
+        }
+        catch (NoSuchFieldException ignored) {
+            //presumably we are on kafka <= 3.7.0, so we can move on with life
+        }
+        catch (IllegalArgumentException | IllegalAccessException | SecurityException e) {
+            System.out.println("tried to fix it, went boom");
```
Comment (k-wall) [inline impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java:509]: we should use a logger.

**Comment by k-wall** [comment]
lgtm. In another place,  https://github.com/ozangunalp/kafka-native/pull/195/files this was the approach that was taken.  But I'm happy with this approach.

File: impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java (hunk 1 above)
Comment (SamBarker) [reply impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java:144]: I'm guessing its intellij specific but still useful to shutup another warning.

**Pull request merged**